bundles/org.openhab.binding.foobot/src/main/java/org/openhab/binding/foobot/internal/handler/FoobotAccountHandler.java

```diff
@@ -0,0 +1,241 @@
+     */
+    private static final long DEVICES_INTERVAL_MINUTES = Duration.ofDays(1).minus(Duration.ofMinutes(1)).toMinutes();
+    private static final Duration SENSOR_INTERVAL_OFFSET_SECONDS = Duration.ofSeconds(15);
+
+    private final Logger logger = LoggerFactory.getLogger(FoobotAccountHandler.class);
+
+    private final FoobotApiConnector connector;
+
+    private String username = "";
+    private int refreshInterval;
+    private @Nullable ScheduledFuture<?> refreshDeviceListJob;
+    private @Nullable ScheduledFuture<?> refreshSensorsJob;
+    private @NonNullByDefault({}) ExpiringCache<List<FoobotDeviceHandler>> dataCache;
+
+    public FoobotAccountHandler(Bridge bridge, FoobotApiConnector connector) {
+        super(bridge);
+        this.connector = connector;
+    }
+
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Collections.singleton(FoobotAccountDiscoveryService.class);
+    }
+
+    public List<FoobotDevice> getDeviceList() throws FoobotApiException {
+        return connector.getAssociatedDevices(username);
+    }
+
+    public int getRefreshInterval() {
+        return refreshInterval;
+    }
+
+    @Override
+    public void initialize() {
+        final FoobotAccountConfiguration accountConfig = getConfigAs(FoobotAccountConfiguration.class);
+        final List<String> missingParams = new ArrayList<>();
+
+        if (StringUtils.trimToNull(accountConfig.apiKey) == null) {
+            missingParams.add("'apikey'");
+        }
+        if (StringUtils.trimToNull(accountConfig.username) == null) {
+            missingParams.add("'username'");
+        }
+
+        if (!missingParams.isEmpty()) {
+            final boolean oneParam = missingParams.size() == 1;
+            final String errorMsg = String.format(
+                    "Parameter%s [%s] %s mandatory and must be configured and not be empty", oneParam ? "" : "s",
+                    StringUtils.join(missingParams, ", "), oneParam ? "is" : "are");
+
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, errorMsg);
+            return;
+        }
+        username = accountConfig.username;
+        connector.setApiKey(accountConfig.apiKey);
+        refreshInterval = accountConfig.refreshInterval;
+        if (this.refreshInterval < 5) {
+            logger.warn(
+                    "Refresh interval time [{}] is not valid. Refresh interval time must be at least 5 minutes. Setting to 7 minutes",
+                    accountConfig.refreshInterval);
+            refreshInterval = DEFAULT_REFRESH_PERIOD_MINUTES;
+        }
+        logger.debug("Foobot Account bridge starting... user: {}, refreshInterval: {}", accountConfig.username,
+                refreshInterval);
+
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_PENDING, "Wait to get associated devices");
```

Comment:
CONFIGURATION_PENDING is not a valid sub-status of OFFLINE, see https://www.openhab.org/docs/concepts/things.html#status-details.